archive/demos/shiny-drum-machine-audio.js

```diff
@@ -0,0 +1,307 @@
+/* eslint require-jsdoc: "off" */
+
+import {INSTRUMENTS, freeze, clone} from './shiny-drum-machine-data.js';
+
+const AudioContext = window.AudioContext ?? window.webkitAudioContext;
+
+const context = new AudioContext();
+
+const LOOP_LENGTH = 16;
+const BEATS_PER_FULL_NOTE = 4;
+const VOLUMES = freeze([0, 0.3, 1]);
+
+
+async function fetchAudio(url) {
```

Comment:
Perhaps `fetchAndDecode` is a better name since it produces a decoded AudioBuffer.

Reply:
Renamed.